### PR TITLE
Update fees.md

### PR DESCRIPTION
**Pull request by NukeManDan**
- fix 404 links
- update to FRAME v2
- formatting (plus some minor in frontier page, thanks to the IDE)
- hard tabs for rust snippets

### CODE REVIEW

File: docs/knowledgebase/runtime/fees.md [hunk 1]
```diff
@@ -315,10 +315,17 @@ impl WeighData<(&u32, &u64)> for CustomWeight {
         ...
     }
 }
+impl ClassifyDispatch<(T)> for CustomWeight {
+    fn classify_dispatch(&self, target: T) -> DispatchClass{
+        ...
+    }
+}
```
[the remark this answers is not in the document]
Comment (NukeManDan) [reply docs/knowledgebase/runtime/fees.md:323]: https://github.com/paritytech/substrate/blob/master/frame/example/src/lib.rs#L489-L494

[the remark this answers is not in the document]
Comment (jimmychu0807) [reply docs/knowledgebase/runtime/fees.md:323]: I believe you almost got it!
Based on [this page](https://substrate.dev/rustdocs/latest/frame_support/weights/index.html), they also need to implement `PaysFee` trait. 

An example is here: https://github.com/paritytech/substrate/blob/master/frame/example/src/lib.rs#L324-L349

[the remark this answers is not in the document]
Comment (jimmychu0807) [reply docs/knowledgebase/runtime/fees.md:323]: Actually, having a second look of the whole article, what you added is already mentioned in L284 - 293 (and the `PaysFee` implementation is also covered). This section is highlighting L313 that `WeightData<...>` is not using a generic type but a concrete `(&u32, u64)`. So this section doesn't need to be changed and is all good.

**Comment by NukeManDan** [comment]
Thanks for the full review @jimmychu0807 ! While we have it fresh, we should update to close #1124 throughout, would you be willing to help on this PR or should we open a followup?

File: docs/knowledgebase/runtime/fees.md [hunk 2]
```diff
@@ -311,26 +311,34 @@ this. Just note that, in that case, your code would roughly look like:
...
+	}
 }
-
-// given dispatch:
-decl_module! {
-    #[weight = CustomWeight]
-    fn foo(a: u32, b: u64) { ... }
+// impl `ClassifyDispatch<(&u32, &u64)>` , `PaysFee<(&u32, &u64)>`...
+
+// example dispatchable:
```
Comment (NukeManDan) [inline docs/knowledgebase/runtime/fees.md:320]: Looks reasonable (with custom weight use)?

**Review by jimmychu0807 — APPROVED** [review]
Looks good to me. Thanks @NukeManDan

**Pull request merged**